src/components/__tests__/FormLogin.cy.js

```diff
@@ -249,24 +261,83 @@ describe('<FormLogin>', () => {
       cy.viewport('macbook-16');
     });
 
-    it('uses the login store', () => {
-      const loginStore = useLoginStore();
-      // initial state
-      expect(loginStore.user.email).to.equal(emptyUser.email);
-      expect(loginStore.user.password).to.equal(emptyUser.password);
-      // type email
-      cy.dataCy('form-login-email').find('input').clear();
-      cy.dataCy('form-login-email').find('input').type(email);
-      // check email in store
-      cy.dataCy('form-login-email').then(() => {
-        expect(loginStore.user.email).to.equal(email);
+    it('allows to save tokens and user into store', () => {
+      const store = useLoginStore();
+      store.setAccessToken(tokenAccess);
+      store.setRefreshToken(tokenRefresh);
+      store.setUser(user);
+      expect(store.getAccessToken).to.equal(tokenAccess);
+      expect(store.getRefreshToken).to.equal(tokenRefresh);
+      expect(store.getUser).to.deep.equal(user);
+    });
+
+    it('shows error if login is called and username is not set', () => {
+      const store = useLoginStore();
+      cy.wrap(store.login({ username: '', password })).then((result) => {
+        expect(result).to.equal(null);
+        cy.get(classSelectorQNotificationMessage)
+          .should('be.visible')
+          .and('contain', i18n.global.t('login.form.messageEmailReqired'));
       });
-      // type password
-      cy.dataCy('form-login-password').find('input').clear();
-      cy.dataCy('form-login-password').find('input').type(password);
-      // check password in store
-      cy.dataCy('form-login-password').then(() => {
-        expect(loginStore.user.password).to.equal(password);
+    });
+
+    it('shows error if login is called and password is not set', () => {
+      const store = useLoginStore();
+      cy.wrap(store.login({ username, password: '' })).then((result) => {
+        expect(result).to.equal(null);
+        cy.get(classSelectorQNotificationMessage)
+          .should('be.visible')
+          .and('contain', i18n.global.t('login.form.messageEmailReqired'));
+      });
+    });
+
+    it('shows error if API call fails (error has message)', () => {
+      const store = useLoginStore();
+      cy.intercept('POST', apiLoginUrl, {
+        statusCode: errorStatusCode,
+      }).then(() => {
+        cy.wrap(store.login({ username, password })).then((result) => {
+          expect(result).to.equal(null);
+          cy.get(classSelectorQNotificationMessage)
+            .should('be.visible')
+            .and('contain', i18n.global.t('login.apiMessageErrorWithMessage'))
+            .and('contain', errorStatusCode);
+        });
+      });
+    });
+
+    it('calls API and set token on successful login', () => {
+      // intercept login API call
+      cy.intercept('POST', apiLoginUrl, {
+        statusCode: 200,
```

Comment:
You can reuse HTTP successful status code defined here https://github.com/auto-mat/ride-to-work-by-bike-frontend/blob/7f03f89b51c32697f66a725d615e3a1703389d16/test/cypress/support/commonTests.ts#L79

Reply:
Fixed with https://github.com/auto-mat/ride-to-work-by-bike-frontend/pull/559/commits/dc97ab064797faf9e2bffe86854b2ffc81eee2c4.